packages/paste-core/utilities/box/src/FilterBoxPropsFromProps.ts

```diff
@@ -0,0 +1,101 @@
+import * as lodash from 'lodash';
+
+const supportedBoxProps = [
+  'margin',
+  'm',
+  'marginTop',
+  'mt',
+  'marginRight',
+  'mr',
+  'marginBottom',
+  'mb',
+  'marginLeft',
+  'ml',
+  'marginX',
+  'mx',
+  'marginY',
+  'my',
+  'padding',
+  'p',
+  'paddingTop',
+  'pt',
+  'paddingRight',
+  'pr',
+  'paddingBottom',
+  'pb',
+  'paddingLeft',
+  'pl',
+  'paddingX',
+  'px',
+  'paddingY',
+  'py',
+  'width',
+  'height',
+  'display',
+  'minWidth',
+  'minHeight',
+  'maxWidth',
+  'maxHeight',
+  'size',
+  'verticalAlign',
+  'overflow',
+  'overflowX',
+  'overflowY',
+  'alignItems',
+  'alignContent',
+  'justifyItems',
+  'justifyContent',
+  'flexWrap',
+  'flexDirection',
+  'flex',
+  'flexGrow',
+  'flexShrink',
+  'flexBasis',
+  'justifySelf',
+  'alignSelf',
+  'order',
+  'backgroundImage',
+  'backgroundSize',
+  'backgroundPosition',
+  'backgroundRepeat',
+  'backgroundColor',
+  'border',
+  'borderWidth',
+  'borderStyle',
+  'borderColor',
+  'borderRadius',
+  'borderTop',
+  'borderTopWidth',
+  'borderTopStyle',
+  'borderTopColor',
+  'borderTopLeftRadius',
+  'borderTopRightRadius',
+  'borderRight',
+  'borderRightWidth',
+  'borderRightStyle',
+  'borderRightColor',
+  'borderBottom',
+  'borderBottomWidth',
+  'borderBottomStyle',
+  'borderBottomColor',
+  'borderBottomLeftRadius',
+  'borderBottomRightRadius',
+  'borderLeft',
+  'borderLeftWidth',
+  'borderLeftStyle',
+  'borderLeftColor',
+  'borderX',
+  'borderY',
+  'borderColor',
+  'boxShadow',
+  'zIndex',
+] as const;
+
+type Exceptions = typeof supportedBoxProps | string[];
+
+export const filterBoxPropsFromProps = (props: {}, exceptions: Exceptions = []): {} => {
+  const boxPropsToFilter = supportedBoxProps.filter(prop => {
```

Comment:
Slight improvement:
```
let boxPropsToFilter = supportedBoxProps;
if (exceptions.length !== 0) {
   boxPropsToFilter = supportedBoxProps.filter(prop => {
      return !exceptions.includes(prop);
  });
}
return lodash.omit(props, boxPropsToFilter);
```

Reply:
I always like Set's for filtering work another implementation

```js
const validProps = new Set(['allThoseProps']);

const incomingAttributes = new Set(Object.entries(attributes));
const whiteList = intersection(validProps, exceptions);

return lodash.pick(attributes, whiteList)

```

Lodash is also removing omit because it's slow AF apparently and recommends pick.

https://github.com/lodash/lodash/issues/2930

Reply:
Thinking about it more and what you said here 

> This isn't something we can just run in dev and remove in prod; if we remove this in prod all the props will fall through unexpectedly.

We should probably throw an error because the props are incorrect and we _want_ to strip this check in production. If it doesn't work locally in JS land I feel as if we have a pretty decent shot at making sure we're not letting bad props in.

```js
const validProps = new Set(['allThoseProps']);

const incomingAttributes = new Set(Object.entries(attributes));
const whiteList = intersection(validProps, exceptions);

if( lodash.omit(attributes, whiteList).length > 0 ){
throw new Error("THOSE AREN'T THE RIGHT PROPS!");
}
```